### PR TITLE
Dependency bumps

### PR DESCRIPTION
**Pull request by stevejalim**
**Prod deps**
* Resolves #14420 Bump boto3 from 1.34.65 to 1.34.79 
* Resolves #14405 Bump pillow from 10.2.0 to 10.3.0
* Resolves #14390 Bump lxml from 5.1.0 to 5.2.0
* Resolves #14387 Bump sentry-sdk from 1.42.0 to 1.44.0 
* Resolves #14380 Bump pygithub from 2.2.0 to 2.3.0
* Resolves #14379 Bump newrelic from 9.7.1 to 9.8.0
* Resolves #14378 Bump jq from 1.6.0 to 1.7.0

**Dev deps**
* Resolves #14422 Bump pipdeptree from 2.16.1 to 2.17.0 
* Resolves #14389 Bump pytest-mock from 3.12.0 to 3.14.0 
* Resolves #14385 Bump ruff from 0.3.3 to 0.3.4
* Resolves #14381 Bump pytest-cov from 4.1.0 to 5.0.0 
* ~Resolves #14375 Bump pytest from 7.4.4 to 8.1.1~

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -65,7 +65,7 @@ repos:
           - --comment-style
           - "|#|"
   - repo: https://github.com/charliermarsh/ruff-pre-commit
```
Comment (robhudson) [inline .pre-commit-config.yaml:67]: We may as well change this to the new URL:
`- repo: https://github.com/astral-sh/ruff-pre-commit`

Comment (robhudson) [reply .pre-commit-config.yaml:67]: Also, v0.3.5 is out now.

File: requirements/dev.in [hunk 2]
```diff
@@ -5,21 +5,21 @@ braceexpand==0.1.7
...
 markdown-it-py>=2.2.0
-pipdeptree==2.16.1
+pipdeptree==2.17.0
 py==1.11.0
 Pygments>=2.15.0  # to bring it up to a secure version
 PyPOM==2.2.4
 pyquery==2.0.0
-pytest==7.4.4  # Included because we use it directly, but also a subdep of pytest-selenium's subdeps
-pytest-cov==4.1.0
+pytest==8.1.1  # Included because we use it directly, but also a subdep of pytest-selenium's subdeps
```
Comment (robhudson) [inline requirements/dev.in:13]: I thought this one caused problems?

**Review by robhudson — APPROVED** [review]
r+wc

File: requirements/dev.in (hunk 2 above)
Comment (stevejalim) [reply requirements/dev.in:13]: I thought 8.0.0 was the one, but happy to roll back to 7.4.4

File: .pre-commit-config.yaml (hunk 1 above)
Comment (stevejalim) [reply .pre-commit-config.yaml]: Updated!

File: requirements/dev.in (hunk 2 above)
Comment (robhudson) [reply requirements/dev.in]: I trust your work and testing, just jumped out at me b/c I thought I tried some 8.x version but can't remember which. If this works then great!

Comment (stevejalim) [reply requirements/dev.in]: well 7.4.4 works fine right now, too. Can bump it back up next time.

**Pull request merged**